### ISSUE TITLE
mapper(u0)  not defined for AttractorsViaFeaturizing

### ISSUE THREAD

**Issue by awage** [JuliaDynamics/Attractors.jl#137]
In the docstring of `AttractorsViaFeaturizing` we have: 
``` julia
This `mapper` also allows the syntax `mapper(u0)` but only if the `grouping_config`
is _not_ `GroupViaClustering`.
```
but the function is not implemented: 
```julia
   function dumb_map(z, p, n)
        x, y = z
        r = p[1]
        if r < 0.5
            return SVector(0.0, 0.0)
        else
            if x ≥ 0
                return SVector(r, r)
            else
                return SVector(-r, -r)
            end
        end
    end

    r = 1.0
    ds = DeterministicIteratedMap(dumb_map, [0., 0.], [r])
    u0s = [1 => [r, r], 2 => [-r, -r]] # template ics

    xg = yg = range(-2.0, 2.0; length=100)
    grid = (xg, yg)
    expected_fs_raw = Dict(1 => 0.5, 1 => 0.5)
    featurizer(A, t) = SVector(A[1][1])

function features_from_u(u)
    A, t = trajectory(ds, 100, u; Ttr = 1000, Δt = 1)
featurizer(A, t)
end
templates = Dict(k => features_from_u(u) for (k, u) in u0s)
config = GroupViaNearestFeature(templates)
mapper = AttractorsViaFeaturizing(ds, featurizer, config; Ttr=500)
mapper([1., 0.])
```
returns: 

ERROR: MethodError: objects of type AttractorsViaFeaturizing{DeterministicIteratedMap{false, SVector{2, Float64}, 2, 
typeof(dumb_map), Vector{Float64}}, GroupViaNearestFeature{1, Float64, NearestNeighbors.KDTree{SVector{1, Float64}, 
Euclidean, Float64, SVector{1, Float64}}, Vector{Int64}}, typeof(featurizer), Int64, 2, Float64} **are not callable**

I will propose a PR to solve this. It is not crucial but it a convenience function that is helpful.

**Comment by Datseris** [comment]
Sorry @awage the docstring is wrong. With `GroupViaNearestFeature` it is not possible to do `mapper(u0)`. There is simply no other nearest feature to map to, so it doesn't make sense. We just need to correct the docstring.

**Comment by awage** [comment]
It is possible since you give the templates to `GroupViaNearestFeature.` : 

`Then, generated features from initial conditions in AttractorsViaFeaturizing are labelled according to the feature in templates that is closest (the label is the index of the closest template).`

So it works in a similar fashion to `AttractorsViaProximity` but with features. In #138 the mapper is functional.

**Comment by Datseris** [comment]
Ah, sorry, I mistaken this with the other version that Kalel added, the pairwise comparison. good.

**Comment by awage** [comment]
Solved in PR #137
